src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -317,8 +317,10 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
                 for lib in libs.iter() {
                     match *lib {
                         LibKind::Dylib => {
-                            if let Ok((prefix, suffix)) = self.dylib(unit.kind) {
-                                ret.push(format!("{}{}{}", prefix, stem, suffix));
+                            match self.dylib(unit.kind) {
+                                Ok((prefix, suffix)) =>
+                                    ret.push(format!("{}{}{}", prefix, stem, suffix)),
+                                Err(e) => try!(self.config.shell().warn(format!("{}", e)))
```

Comment:
Ah here we probably don't want to issue a warning because there's not much we can really do about it anyway (other than just silently dropping it)